cloud/testserver/deployments.go

```diff
@@ -7,160 +7,151 @@ import (
 	"encoding/json"
 	"io"
 	"net/http"
-	"strings"
-	"sync"
-	"sync/atomic"
 
 	"github.com/julienschmidt/httprouter"
 	"github.com/terramate-io/terramate/cloud"
 	"github.com/terramate-io/terramate/cloud/deployment"
+	"github.com/terramate-io/terramate/cloud/testserver/cloudstore"
+	"github.com/terramate-io/terramate/errors"
 )
 
-type deploymentHandler struct {
-	mu          sync.Mutex
-	nextStackID int64
-	// as hacky as it can get:
-	// map of organization -> (map of deployment_id -> (map of stack_id -> deployment))
-	deployments map[string]map[string]map[int64]cloud.DeploymentStackRequest
+// GetDeployments is the GET /deployments handler.
+func GetDeployments(store *cloudstore.Data, w http.ResponseWriter, _ *http.Request, p httprouter.Params) {
+	orguuid := cloud.UUID(p.ByName("orguuid"))
+	deployuuid := cloud.UUID(p.ByName("deployuuid"))
 
-	events map[string]map[string]map[string][]string
-}
+	org, found := store.GetOrg(orguuid)
+	if !found {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
 
-func newDeploymentEndpoint() *deploymentHandler {
-	return &deploymentHandler{
-		deployments: make(map[string]map[string]map[int64]cloud.DeploymentStackRequest),
-		events:      make(map[string]map[string]map[string][]string),
+	deploymentInfo, found := store.GetDeployment(&org, deployuuid)
+	if !found {
+		w.WriteHeader(http.StatusNotFound)
+		return
 	}
+	w.Header().Add("Content-Type", "application/json")
+	marshalWrite(w, deploymentInfo)
 }
 
-func (dhandler *deploymentHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	dhandler.mu.Lock()
-	defer dhandler.mu.Unlock()
-
-	params := httprouter.ParamsFromContext(r.Context())
-	orguuid := params.ByName("orguuid")
-	deployuuid := params.ByName("deployuuid")
-
-	if !strings.HasPrefix(r.Header.Get("User-Agent"), "terramate/") {
-		w.WriteHeader(http.StatusBadRequest)
+// CreateDeployment is the POST /deployments handler.
+func CreateDeployment(store *cloudstore.Data, w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	defer func() { _ = r.Body.Close() }()
+	data, _ := io.ReadAll(r.Body)
+	var rPayload cloud.DeploymentStacksPayloadRequest
+	err := json.Unmarshal(data, &rPayload)
+	if err != nil {
```

Comment:
nitpick: I think this has the added benefit of not mistakenly reusing the `err` variable later in the function.
 
```suggestion
	if err := json.Unmarshal(data, &rPayload); err != nil {
```